airflow-core/src/airflow/utils/hitl_shared_links.py

```diff
@@ -0,0 +1,220 @@
+
+    def __init__(self):
+        self.secret_key = conf.get("api", "hitl_shared_link_secret_key", fallback="")
+        self.default_expiration_hours = conf.getint("api", "hitl_shared_link_expiration_hours", fallback=24)
+
+    def is_enabled(self) -> bool:
+        """Check if HITL shared links are enabled."""
+        return conf.getboolean("api", "hitl_enable_shared_links", fallback=False)
+
+    def _generate_signature(self, payload: str) -> str:
+        """Generate HMAC signature for the payload."""
+        if not self.secret_key:
+            raise ValueError("HITL shared link secret key is not configured")
+
+        signature = hmac.new(
+            self.secret_key.encode("utf-8"), payload.encode("utf-8"), hashlib.sha256
+        ).digest()
+        return base64.urlsafe_b64encode(signature).decode("utf-8")
+
+    def _verify_signature(self, payload: str, signature: str) -> bool:
+        """Verify HMAC signature for the payload."""
+        expected_signature = self._generate_signature(payload)
+        return hmac.compare_digest(expected_signature, signature)
+
+    def generate_link(
+        self,
+        dag_id: str,
+        dag_run_id: str,
+        task_id: str,
+        map_index: int | None = None,
+        link_type: str = "action",
+        action: str | None = None,
+        expires_in_hours: int | None = None,
+        base_url: str | None = None,
+    ) -> dict[str, Any]:
+        """
+        Generate a shared link for HITL task.
+
+        :param dag_id: DAG ID
+        :param dag_run_id: DAG run ID
+        :param task_id: Task ID
+        :param map_index: Map index for mapped tasks
+        :param link_type: Type of link ('action' or 'redirect')
+        :param action: Action to perform (for action links)
+        :param expires_in_hours: Custom expiration time in hours
+        :param base_url: Base URL for the link
+        """
+        if not self.is_enabled():
+            raise ValueError("HITL shared links are not enabled")
+
+        if link_type == "action" and not action:
+            raise ValueError("Action is required for action-type links")
+
+        expiration_hours = expires_in_hours or self.default_expiration_hours
+        expires_at = timezone.utcnow() + timedelta(hours=expiration_hours)
+
+        payload_data = {
+            "dag_id": dag_id,
+            "dag_run_id": dag_run_id,
+            "task_id": task_id,
+            "map_index": map_index,
+            "link_type": link_type,
+            "action": action,
+            "expires_at": expires_at.isoformat(),
+        }
+
+        payload_str = json.dumps(payload_data, sort_keys=True)
+        signature = self._generate_signature(payload_str)
+
+        encoded_payload = base64.urlsafe_b64encode(payload_str.encode("utf-8")).decode("utf-8")
+
+        if base_url is None:
+            base_url = conf.get("api", "base_url", fallback="http://localhost:8080")
```

Comment:
I don't think we should have a fallback here. If it's not set, we should probably fail?

Reply:
Yes, I think failing in this case would be better than failing. Changed it.